### PR TITLE
feat(Core/Commands): Deserter command (BG/Instance)

### PR DESCRIPTION
**Pull request by poszer**
Implement deserter command (For sure, a lot people using this with command .unaura to remove deserter debuff from players - Including me too :rofl: ).

##### CHANGES PROPOSED:

- Adding .deserter command
- Subcommands ;
* instance add $time (Adds instance deserter debuff with $time duration on target)
* instance remove (Removes the instance deserter debuff)
* bg add $time (Adds BG deserter debuff with $time duration on target)
* bg remove (Removes the bg deserter debuff)

###### ISSUES ADDRESSED:

Closes 

##### TESTS PERFORMED:

Yes, builded without errors. Tested in game, all commands work without problem.

##### HOW TO TEST THE CHANGES:

1. Try these commands :

- .deserter instance add 3600

- .deserter bg add 3600

- .deserter instance remove

- .deserter bg remove

Time is in seconds, 3600 = 1hour

##### KNOWN ISSUES AND TODO LIST:

- [ ]
- [ ] 

##### Target branch(es):

Master

### CODE REVIEW

**Comment by Viste** [comment]
`#include "SpellAuraEffects.h"`

**Comment by poszer** [comment]
> ```
> #include "SpellAuras.h"
> #include "SpellAuraEffects.h"
> ```

Thanks, will try

File: src/server/scripts/Commands/cs_deserter.cpp [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+            handler->SetSentErrorMessage(true);
+            return false;
+         }
+        aura->SetDuration(time * IN_MILLISECONDS);
+
+         return true;
+    }
+
+    static bool HandleDeserterRemove(ChatHandler* handler, char const* args, bool isInstance)
+    {
```
Comment (Viste) [inline src/server/scripts/Commands/cs_deserter.cpp]:
```
static bool HandleDeserterRemove(ChatHandler* handler, bool isInstance)
```
you add args but dont use it travis will fail

File: src/server/scripts/Commands/cs_deserter.cpp [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    }
+
+    static bool HandleDeserterBGAdd(ChatHandler* handler, char const* args)
+    {
+        return HandleDeserterAdd(handler, args, false);
+    }
+
+    static bool HandleDeserterInstanceRemove(ChatHandler* handler, char const* args)
+    {
+        return HandleDeserterRemove(handler, args, true);
```
Comment (Viste) [inline src/server/scripts/Commands/cs_deserter.cpp]:
```
return HandleDeserterRemove(handler, true);
```

File: src/server/scripts/Commands/cs_deserter.cpp [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    }
+
+    static bool HandleDeserterBGRemove(ChatHandler* handler, char const* args)
+    {
+        return HandleDeserterRemove(handler, args, false);
```
Comment (Viste) [inline src/server/scripts/Commands/cs_deserter.cpp]:
```
return HandleDeserterRemove(handler, true);
```

**Comment by BarbzYHOOL** [comment]
Nice work

**Comment by comix1988** [comment]
all commands works, both on themselves and on a player.

good job @poszer

**Comment by Viste** [comment]
@FrancescoBorzi need some approve here

**Pull request merged**